contracts/strategies/QVSimple.sol

```diff
@@ -0,0 +1,270 @@
+    function _checkOnlyAfterAllocation() internal view virtual {
+        if (block.timestamp <= allocationEndTime) revert ALLOCATION_NOT_ENDED();
+    }
+
+    /// @notice Distribute the tokens to the recipients
+    /// @dev The '_sender' must be a pool manager and the allocation must have ended
+    /// @param _recipientIds The recipient ids
+    /// @param _sender The sender of the transaction
+    function _distribute(address[] memory _recipientIds, bytes memory, address _sender)
+        internal
+        virtual
+        override
+        onlyPoolManager(_sender)
+        onlyAfterAllocation
+    {
+        uint256[] memory payouts = _votingState._getPayout(_recipientIds, poolAmount);
+
+        for (uint256 i; i < payouts.length;) {
+            address recipientId = _recipientIds[i];
+            Recipient storage recipient = _recipients[recipientId];
+
+            uint256 amount = payouts[i];
+
+            if (paidOut[recipientId] || !_isAcceptedRecipient(recipientId) || amount == 0) {
+                revert RECIPIENT_ERROR(recipientId);
+            }
+
+            IAllo.Pool memory pool = allo.getPool(poolId);
+            _transferAmount(pool.token, recipient.recipientAddress, amount);
+
+            paidOut[recipientId] = true;
+
+            emit Distributed(recipientId, abi.encode(recipient.recipientAddress, amount, _sender));
+            unchecked {
+                ++i;
+            }
+        }
+        if (!distributionStarted) {
+            distributionStarted = true;
+        }
+    }
+
+    /// @notice Allocate voice credits to an array of recipients
+    /// @param __recipients The recipients
+    /// @param _amounts The amounts of voice credits to allocate
+    /// @param _data The data
+    /// @param _sender The actual sender of the transaction
+    function _allocate(address[] memory __recipients, uint256[] memory _amounts, bytes memory _data, address _sender)
+        internal
+        override
+    {
+        uint256 voiceCreditsAlreadyAllocated = voiceCreditsAllocated[_sender];
+
+        // check that the sender can allocate votes
+        if (!_isValidAllocator(_sender)) revert UNAUTHORIZED();
+
+        uint256 voiceCreditsToAllocate;
+
+        for (uint256 i = 0; i < __recipients.length; i++) {
+            // check the voice credits to allocate is > 0
+            if (_amounts[i] == 0) revert INVALID();
+
+            // check that the recipient is accepted
+            if (!_isAcceptedRecipient(__recipients[i])) revert RECIPIENT_ERROR(__recipients[i]);
+
+            // sum up the voice credits to allocate
+            voiceCreditsToAllocate += _amounts[i];
+
+            emit Allocated(__recipients[i], _sender, voiceCreditsToAllocate, _data);
+        }
+
+        // check that the allocator has voice credits left to allocate
+        if (!_hasVoiceCreditsLeft(voiceCreditsToAllocate, voiceCreditsAlreadyAllocated)) revert INVALID();
+
+        _votingState._voteWithVoiceCredits(__recipients, _amounts);
```

Comment:
In hindsight I wonder if QVHelper and QFHelper libraries should also include functions that don't have arrays but single elements as input.

Reply:
We can add them for convenience, users can still just pass an array with 1 element. We can add this in to the libraries in an other PR